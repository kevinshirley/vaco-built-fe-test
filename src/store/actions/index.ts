export function createTypes(base: string, types: string[]) {
  const res: any = {};
  types.forEach(type => {
    res[type] = `${base}_${type}`;
  });
  return res;
}

export const action = (type: string, payload = {}, meta = {}) => ({
  type,
  payload,
  meta,
});

export const actionWithoutStoreUpdate = (type: string, payload = {}, meta = {}) => ({
  type,
  payload,
  meta,
  shouldNotUpdateStore: true,
});

export const ROOT = createTypes('ROOT', [
  'INITIAL_LOAD',
]);

export const STOCKS = createTypes('STOCKS', [
  'GET_COMPANY_PROFILE',
  'SET_COMPANY_PROFILE',
  'RESET_COMPANY_PROFILE',
  'SET_STOCK_SYMBOLS',
  'SYMBOL_INPUT_RESULTS',
  'SET_SEARCH_INPUT_RESULTS',
  'RESET_SEARCH_INPUT_RESULTS',
  'SET_MARKET_NEWS',
]);

export const root = {
  initialLoad: (params: any) => actionWithoutStoreUpdate(ROOT.INITIAL_LOAD, { params }),
};

export const stocks = {
  getCompanyProfile: (payload: { symbol: string }) => action(STOCKS.GET_COMPANY_PROFILE, payload),
  setCompanyProfile: (payload: any) => action(STOCKS.SET_COMPANY_PROFILE, payload),
  resetCompanyProfile: () => action(STOCKS.RESET_COMPANY_PROFILE),
  setStockSymbols: (payload: any) => action(STOCKS.SET_STOCK_SYMBOLS, payload),
  symbolInputResults: (payload: any) => action(STOCKS.SYMBOL_INPUT_RESULTS, payload),
  setSearchInputResults: (payload: any) => action(STOCKS.SET_SEARCH_INPUT_RESULTS, payload),
  resetSearchInputResults: () => action(STOCKS.RESET_SEARCH_INPUT_RESULTS),
  setMarketNews: (payload: any) => action(STOCKS.SET_MARKET_NEWS, payload),
};
