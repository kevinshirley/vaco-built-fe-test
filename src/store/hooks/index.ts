import { useDispatch } from 'react-redux';
import { useCallback } from 'react';

/**
 * Creates an action dispatcher.
 *
 * Uses like:
 * ```js
 * const action = useAction(actions.myAction);
 * // ...
 * <button onClick={action}>Call action</button>
 * ```
 */

export function useAction(action: any) {
  const dispatch = useDispatch();

  return useCallback(
    (...args: any) => {
      dispatch(action(...args));
    },
    [action, dispatch],
  );
}
