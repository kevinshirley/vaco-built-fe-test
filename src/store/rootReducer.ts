import { combineReducers } from 'redux';

import stock from './reducers/stock';
import stocks from './reducers/stocks';
import marketNews from './reducers/news';

export default combineReducers({
  stock,
  stocks,
  marketNews,
});
