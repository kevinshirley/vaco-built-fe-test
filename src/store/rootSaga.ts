import { all } from 'redux-saga/effects';
import 'isomorphic-unfetch';
import { 
  watchGetCompanyProfile,
  watchGetStockSymbols,
  watchSymbolInputResults,
  watchStockMarketNews,
} from '../store/sagas/stocks';

function* rootSaga() {
  yield all([
    watchGetCompanyProfile(),
    watchGetStockSymbols(),
    watchSymbolInputResults(),
    watchStockMarketNews(),
  ]);
}

export default rootSaga;
