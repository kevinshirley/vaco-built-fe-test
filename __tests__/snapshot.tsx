import { render } from '@testing-library/react'
import Home from '../pages'
import { wrapper } from '../src/store'

it('renders homepage unchanged', () => {
  const HomeWithRedux = wrapper.withRedux(Home)
  const { container } = render(<HomeWithRedux />)
  expect(container).toMatchSnapshot()
})