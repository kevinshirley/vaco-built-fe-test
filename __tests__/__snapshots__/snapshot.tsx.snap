// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="c-home"
    data-testid="home-component"
  >
    <div
      class="c-home__search-stocks"
    >
      <div
        class="c-home__search-wrapper"
      >
        <div
          class="c-search"
        >
          <div
            class="MuiFormControl-root MuiTextField-root c-search__field css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                aria-label="search"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r0:"
                placeholder="Search stocks"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    â€‹
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="c-search__icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            class="c-search__icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c-home__no-input"
    >
      <h3>
        Please enter a stock symbol to start
      </h3>
    </div>
  </div>
</div>
`;
